upgrade/pack_builder/ImageSignerEcDsa.hpp

```diff
@@ -15,27 +16,55 @@ namespace application
     {
     public:
         ImageSignerEcDsa(hal::SynchronousRandomDataGenerator& randomDataGenerator, infra::ConstByteRange publicKey, infra::ConstByteRange privateKey);
+        virtual ~ImageSignerEcDsa() = default;
 
-        uint16_t SignatureMethod() const override;
-        uint16_t SignatureLength() const override;
         std::vector<uint8_t> ImageSignature(const std::vector<uint8_t>& image) override;
         bool CheckSignature(const std::vector<uint8_t>& signature, const std::vector<uint8_t>& image) override;
 
-        static const uint16_t signatureMethod = 1;
-        static const size_t keyLength = 224;
-
     private:
-        void CalculateSha256(const std::vector<uint8_t>& image);
-        void CalculateSignature(const std::vector<uint8_t>& image);
+        virtual uECC_Curve GetCurve() const = 0;
 
+        void CalculateSha256(const std::vector<uint8_t>& image);
+        void CalculateSignature();
         static int RandomNumberGenerator(uint8_t* dest, unsigned size);
+        static hal::SynchronousRandomDataGenerator* randomDataGenerator;
```

Comment:
I can't make the edits here, but please move randomDataGenerator to the variables section, and give it a dedicated "private" access specifier